packages/reakit/src/Popover/PopoverState.ts

```diff
@@ -163,7 +166,11 @@ export function usePopoverState(
         positionFixed: fixed,
         modifiers: {
           applyStyle: { enabled: false },
-          flip: { enabled: flip, padding: 16 },
+          flip: {
+            enabled: Boolean(flip),
+            padding: 16,
+            ...(typeof flip !== "boolean" ? flip : {})
```

Comment:
could be `...(typeof flip === "object" ? flip : {})`?

Reply:
probably clearer :+1:

Reply:
A small caveat is that `typeof someArray === 'object'`

Reply:
of course _everything is an object_ :joy:

Reply:
🕺

Reply:
`typeof flip === "object" && !Array.isArray(flip) ? flip : {}`